http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -958,10 +958,11 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift: Parallel](
 
       deleteDisksInParallel = List(deleteDisk, deletePostgresDisk).parSequence_
 
-      // TODO: can some of this be done in parallel?
+      // The app must be deleted before the nodepool and disk, to future proof against the app potentially flushing the postgres db somewhere
       task = for {
         _ <- deleteApp
         _ <- deleteNodepool
+        _ <- if (!errorAfterDelete) appQuery.markAsDeleted(msg.appId, ctx.now).transaction.void else F.unit
         _ <- deleteDisksInParallel
```

Comment:
because this is blocking in terms of cluster actions until the app is marked as deleted, if the user requests a disk delete, we don't wait for that portion.